crates/biome_css_analyze/src/lint/nursery/no_duplicate_at_import_rules.rs

```diff
@@ -0,0 +1,122 @@
+    type Query = Ast<CssRuleList>;
+    type State = CssImportAtRule;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        // let mut import_url_list = HashSet::new();
+        let mut import_url_map: HashMap<String, HashSet<String>> = HashMap::new();
+        for rule in node {
+            match rule {
+                AnyCssRule::CssAtRule(item) => match item.rule().ok()? {
+                    AnyCssAtRule::CssImportAtRule(import_rule) => {
+                        let import_url = import_rule.url().ok()?.text().to_lowercase();
+                        if let Some(media_query_set) = import_url_map.get_mut(&import_url) {
+                            // if the current import_rule has no media queries or there are no queries safed in the
+                            // media_query_set, this is always a duplicate
+                            if import_rule.media().text().is_empty() || media_query_set.is_empty() {
+                                return Some(import_rule);
+                            }
+
+                            for media in import_rule.media() {
+                                match media {
+                                    Ok(media) => {
+                                        if !media_query_set.insert(media.text().to_lowercase()) {
+                                            return Some(import_rule);
+                                        }
+                                    }
+                                    _ => return None,
+                                }
+                            }
+                        } else {
+                            let mut media_set: HashSet<String> = HashSet::new();
+                            for media in import_rule.media() {
+                                match media {
+                                    Ok(media) => {
+                                        media_set.insert(media.text().to_lowercase());
+                                    }
+                                    _ => return None,
+                                }
+                            }
+                            import_url_map.insert(import_url, media_set);
+                        }
+                    }
+                    _ => return None,
+                },
+                _ => return None,
+            }
+        }
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, node: &Self::State) -> Option<RuleDiagnostic> {
+        let span = node.range();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                span,
+                markup! {
+                    "Duplicate @import rule."
+                },
+            )
+            .note(markup! {
+                    "Looks like you imported the same file twice. Consider removing one to get rid of this error!"
+            }),
```

Comment:
We should follow rule pillars for diagnostics.
https://biomejs.dev/linter/#rule-pillars